### PR TITLE
base stuff for cursor line position and splitting and stuff

### PR DESCRIPTION
**Pull request by shelvacu**

### CODE REVIEW

File: src/github.com/Virepri/Shoraldele/Buffer/buffer.go [hunk 1]
```diff
@@ -34,3 +36,43 @@ func GetBufferContents(offset int, length int) []byte {
...
+}
+
+func SetCursorPosition(newpos uint64) {
+	cursorPosition = newpos
+	//TODO: Call some function to notify output module
+}
+
+//Wraps at 80 chars, hardcoded 'cause codeday
+//Each element is [length, offset]
+func BufferLineSplit() (res [][2]int) {
```
Comment (RobertCochran) [inline src/github.com/Virepri/Shoraldele/Buffer/buffer.go:51]: Creative; not the way I would have done it. But this keeps the duplication down. I like it.

Comment (shelvacu) [reply src/github.com/Virepri/Shoraldele/Buffer/buffer.go:51]: is creative an insult or a compliment in this case

File: src/github.com/Virepri/Shoraldele/Buffer/buffer.go [hunk 2]
```diff
@@ -2,6 +2,8 @@ package buffer
 
 var work_buffer []byte
 
+var cursorPosition int
```
Comment (RobertCochran) [inline src/github.com/Virepri/Shoraldele/Buffer/buffer.go:5]: If the getter and setter take uint64 args, this should probably be uint64 too

Comment (shelvacu) [reply src/github.com/Virepri/Shoraldele/Buffer/buffer.go:5]: oh whoops

Comment (shelvacu) [reply src/github.com/Virepri/Shoraldele/Buffer/buffer.go:5]: fixed, want everything as ints for codedayness

File: src/github.com/Virepri/Shoraldele/Buffer/buffer.go [hunk 3]
```diff
@@ -34,3 +36,43 @@ func GetBufferContents(offset int, length int) []byte {
 func GetBufferSize() int {
```
Comment (RobertCochran) [inline src/github.com/Virepri/Shoraldele/Buffer/buffer.go:36]: and while we're here, might as well change all the offset args to be uint64

Comment (RobertCochran) [reply src/github.com/Virepri/Shoraldele/Buffer/buffer.go:36]: or maybe not, meh... TODO I guess

**Review by RobertCochran — APPROVED** [review]
Looks fine. 1+

**Pull request merged**